build-logic/settings.gradle.kts

```diff
@@ -1,6 +1,6 @@
 dependencyResolutionManagement {
     versionCatalogs {
-        val libs by creating {
```

Comment:
is this syntax not recommended or something?

Reply:
It leads to warnings about not being used which pollute the output. I think this syntax is fine if the property is actually used.

Reply:
heh, looks like people have been complaining about this for a while https://youtrack.jetbrains.com/issue/KT-38871/Kotlin-Gradle-DSL-MPP-UNUSEDVARIABLE-when-configuring-a-sourceset-with-delegated-property. I'd rather use a consistent syntax then change it up to just avoid a warning. What about just putting a `@file:Suppress("UNUSED_VARIABLE")` at the top of the file?

Reply:
It actually looks like this was [addressed in Kotlin 1.9](https://github.com/jetbrains/kotlin/commit/681e85eaed8e047ea955cbc8b3ea5037b1581cf9) so Gradle 8.3 should fix this. 

I'll test with the rc, and if that actually fixes it, I'll suppress these and make a follow up PR to remove them once 8.3 is released.